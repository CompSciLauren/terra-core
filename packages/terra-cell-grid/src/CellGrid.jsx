import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './CellGrid.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * The children passed to the component
   */
  children: PropTypes.node.isRequired,
  /**
   * Indicates the desired divider styles to apply to a cell grid and its children.
   * One of `'none'`, `'vertical'`, `'horizontal'`, `'both'`.
   */
  dividerStyle: PropTypes.oneOf(['none', 'vertical', 'horizontal', 'both']),
  /**
  /**
   * Function callback for the ref of the cell grid.
   */
  refCallback: PropTypes.func,
};

const defaultProps = {
  dividerStyle: 'none',
};

const CellGrid = ({
  children,
  dividerStyle,
  refCallback,
  ...customProps
}) => {
  const cellGridClassNames = cx([
    `divider-${dividerStyle}`,
    'cell-grid',
    customProps.className,
  ]);

  return (
    <div {...customProps} className={cellGridClassNames} ref={refCallback}>
      {children}
    </div>
  );
};

CellGrid.propTypes = propTypes;
CellGrid.defaultProps = defaultProps;

export default CellGrid;
