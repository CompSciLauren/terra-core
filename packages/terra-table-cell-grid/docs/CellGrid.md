# Cell Grid

The CellGrid is a sub component that groups table cells passed through children. Each row handles it's own selection state through props. The onSelect function that can be passed to the cell grid will return the metaData associated to the cell grid. The metaData allows for greater control of state with the ability to associate any indicator or function with the cell grid.

## Usage

```jsx
import React from 'react';
import { CellGrid, Cell } from 'terra-table-cell-grid';

  <CellGrid
    key="unique-react-key"
    isSelectable
    isSelected
    metaData={{ myInfo: 'my-data' }}
    onSelect={this.handleRowSelection}
  >
    <Cell>
      Cell Content
    </Cell>
    <Cell>
      Cell Content
    </Cell>
  </CellGrid>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
