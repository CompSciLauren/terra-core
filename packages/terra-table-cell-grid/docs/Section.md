# Section

The Terra TableCellGrid Section is a sub component that creates a section header based on the given props and groups the child rows beneath it. The section has the ability to toggle the display of its child cell grids given the isCollapsible and isCollapsed props. For a section to be collapsed both isCollapsible and isCollapsed must be true. Each section handles its collapsible and collapsed state.

## Usage

```jsx
import React from 'react';
import { CellGrid, Section } from 'terra-table-cell-grid';

  <Section
    key="unique-react-key"
    title="Section Title"
    isCollapsible
    isCollapsed
    metaData={{ myInfo: 'my-data' }}
    onSelect={this.handleSelection}
  >
    <CellGrid />
    <CellGrid />
    <CellGrid />
  </Section>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
