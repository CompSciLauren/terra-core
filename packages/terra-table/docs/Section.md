# Section

The Terra Table Section is a sub component that creates a section header based on the given props and groups the child rows beneath it. The section has the ability to toggle the display of its child rows given the isCollapsible and isCollapsed props. For a section to be collapsed both isCollapsible and isCollapsed must be true. Each section handles its collapsible and collapsed state.

## Usage

```jsx
import React from 'react';
import { Row, Section } from 'terra-table';

  <Section
    key="unique-react-key"
    title="Section Title"
    isCollapsible
    isCollapsed
    metaData={{ myInfo: 'my-data' }}
    onSelect={this.handleSelection}
  >
    <Row />
    <Row />
    <Row />
  </Section>
```

## Component Features
* [Cross-Browser Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#cross-browser-support)
* [Responsive Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#responsive-support)
* [Mobile Support](https://github.com/cerner/terra-ui/blob/master/src/terra-dev-site/contributing/ComponentStandards.e.contributing.md#mobile-support)
