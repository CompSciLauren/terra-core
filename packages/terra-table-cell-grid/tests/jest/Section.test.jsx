import React from 'react';
import Section from '../../src/Section';
import CellGrid from '../../src/CellGrid';

// Snapshot test
it('should render a default section', () => {
  const section = shallow(
    <Section colSpan={1} title="test" />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a section with a row', () => {
  const section = shallow(
    <Section colSpan={1} title="test">
      <CellGrid />
    </Section>,
  );
  expect(section).toMatchSnapshot();
});

it('should render a section with isCollpased', () => {
  const section = shallow(
    <Section colSpan={1} title="test" isCollapsed />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a section with isCollapsible', () => {
  const section = shallow(
    <Section colSpan={1} title="test" isCollapsible />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a section with level', () => {
  const section = shallow(
    <Section colSpan={1} title="test" level={3} />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a section with metaData', () => {
  const section = shallow(
    <Section colSpan={1} title="test" metaData={{}} />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a section with onSelect', () => {
  const section = shallow(
    <Section colSpan={1} title="test" onSelect={jest.fn()} />,
  );
  expect(section).toMatchSnapshot();
});

it('should render a section with refCallback', () => {
  const section = shallow(
    <Section colSpan={1} title="test" refCallback={jest.fn()} />,
  );
  expect(section).toMatchSnapshot();
});
