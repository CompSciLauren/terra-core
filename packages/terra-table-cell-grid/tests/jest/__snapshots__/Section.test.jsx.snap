// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default section 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    title="test"
  />
</Fragment>
`;

exports[`should render a section with a row 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    title="test"
  />
  <CellGrid
    dividerStyle="none"
    isSelectable={false}
    isSelected={false}
    isStriped={false}
    selectionStyle="default"
  >
    <Cell />
  </CellGrid>
</Fragment>
`;

exports[`should render a section with isCollapsible 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={true}
    level={2}
    title="test"
  />
</Fragment>
`;

exports[`should render a section with isCollpased 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={true}
    isCollapsible={false}
    level={2}
    title="test"
  />
</Fragment>
`;

exports[`should render a section with level 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={3}
    title="test"
  />
</Fragment>
`;

exports[`should render a section with metaData 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    metaData={Object {}}
    title="test"
  />
</Fragment>
`;

exports[`should render a section with onSelect 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    onSelect={[MockFunction]}
    title="test"
  />
</Fragment>
`;

exports[`should render a section with refCallback 1`] = `
<Fragment>
  <SectionHeader
    isCollapsed={false}
    isCollapsible={false}
    level={2}
    refCallback={[MockFunction]}
    title="test"
  />
</Fragment>
`;
